etc/test-helper.js

```diff
@@ -0,0 +1,92 @@
+const expect = require('expect.js');
+const simple = require('simple-mock');
+const fs = require('fs');
+
+// Placeholders for stubbing built-in Sync Gateway support functions.
+// More info: http://developer.couchbase.com/mobile/develop/guides/sync-gateway/sync-function-api-guide/index.html
+var requireAccess;
+var channel;
+
+var syncFunction;
+
+function init(syncFunctionPath) {
+  // Load the contents of the sync function file into a global variable called syncFunction
+  eval('syncFunction = ' + fs.readFileSync(syncFunctionPath).toString());
+  requireAccess = simple.stub();
+  channel = simple.stub();
+}
+
+function verifyDocumentAccepted(doc, oldDoc, expectedChannels) {
+  syncFunction(doc, oldDoc);
+
+  expect(requireAccess.callCount).to.equal(1);
+  expect(requireAccess.calls[0].arg).to.eql(expectedChannels);
+
+  expect(channel.callCount).to.equal(1);
+  expect(channel.calls[0].arg).to.contain(expectedChannels);
+}
+
+function verifyDocumentCreated(doc, oldDoc) {
+  verifyDocumentAccepted(doc, oldDoc, 'add');
+}
+
+function verifyDocumentReplaced(doc, oldDoc) {
+  verifyDocumentAccepted(doc, oldDoc, 'replace');
+}
+
+function verifyDocumentDeleted(doc, oldDoc) {
+  verifyDocumentAccepted(doc, oldDoc, 'remove');
+}
+
+function verifyDocumentRejected(doc, oldDoc, docType, expectedErrorMessages, expectedChannels) {
+  expect(syncFunction).withArgs(doc, oldDoc).to.throwException(function(ex) {
+    verifyValidationErrors(docType, expectedErrorMessages, ex);
+  });
+
+  expect(requireAccess.callCount).to.equal(1);
+  expect(requireAccess.calls[0].arg).to.eql(expectedChannels);
+
+  expect(channel.callCount).to.equal(0);
+}
+
+function verifyDocumentNotCreated(doc, oldDoc, docType, expectedErrorMessages) {
+  verifyDocumentRejected(doc, oldDoc, docType, expectedErrorMessages, 'add');
+}
+
+function verifyDocumentNotReplaced(doc, oldDoc, docType, expectedErrorMessages) {
+  verifyDocumentRejected(doc, oldDoc, docType, expectedErrorMessages, 'replace');
+}
+
+function verifyDocumentNotDeleted(doc, oldDoc, docType, expectedErrorMessages) {
+  verifyDocumentRejected(doc, oldDoc, docType, expectedErrorMessages, 'remove');
+}
+
+function verifyValidationErrors(docType, expectedErrorMessages, exception) {
```

Comment:
Minor: given that this is a helper module, you may want to validate the parameters. For example, you may want to validate that `expectedErrorMessages` is an array and that `exception` is indeed an exception. This is only so that the caller isn't confused with a callstack if they specify the arguments incorrectly.